### ISSUE TITLE
Extract config management API to another application

### ISSUE THREAD

**Issue by rfratto** [grafana/agent#870]
The configuration management API of the scraping service is very limited in general usefulness, and would be better served to be extracted to its own project. Support for remote management can be replaced by #869. 

Design of this service is out of scope for this issue, but hypothetically, a "configuration management service" could expose a URL for Agents to retrieve their configs from. The configuration management service could also discover Agents and call `/-/reload` against them whenever Agents should pull a new config. This behavior would be nearly identical to the user-facing behavior seen today with the internal config management API.

This depends on #869, and loosening the restriction that prevents scraping service users from using local scrape configs for sharding.

**Comment by rfratto** [comment]
We have no plans to do this; closing as won't do.
